File: world/credit.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
void checksum(int num[], int len)
{
    int sum = 0;
    for (int i = 0; i <= len; i++)
    {
        if (i % 2 == 1)
        {
            sum += 2 * num[i] % 10;
            sum += 2 * num[i] / 10;
        }
        else
        {
            sum += num[i];
        }
    }
    if (sum % 10 != 0)
    {
        printf("INVALID\n");
        exit(0);
    }
}
int main(void)
{
    long n;
    int len = -1;
    int cn[20] = {0};
    do
    {
        n = get_long("Number: ");
    }
    while (n == LONG_MAX);
    for (int i = 0; n != 0; i++)
    {
        cn[i] = n % 10;
        n = n / 10;
        len++;
    }
    checksum(cn, len);
    if (cn[len] == 3 && (cn[len - 1] == 4 || cn[len - 1] == 7) && len == 14)
    {
        printf("AMEX\n");
    }
    else if (cn[len] == 5 && cn[len - 1] >= 1 && cn[len - 1] <= 5 && len == 15)
    {
        printf("MASTERCARD\n");
    }
    else if (cn[len] == 4 && (len == 12 || len == 15))
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

File: sentimental-credit/credit.py
from cs50 import get_int, get_string
import re

while True:
    cn = get_string("Number: ")
    if not re.search("\D", cn):
        break

cn = str(cn)[::-1]
sum = 0
for i in range(len(cn)):
    if i % 2 == 1:
        sum += 2*int(cn[i]) % 10
        sum += int(2*int(cn[i])/10)
    else:
        sum += int(cn[i])
if sum % 10 != 0:
    print("INVALID")
    quit()
if cn[-1] == "3" and (cn[-2] == "4" or cn[-2] == "7") and len(cn) == 15:
    print("AMEX")
elif cn[-1] == "5" and int(cn[-2]) >= 1 and int(cn[-2]) <= 5 and len(cn) == 16:
    print("MASTERCARD")
elif cn[-1] == "4" and (len(cn) == 13 or len(cn) == 16):
    print("VISA")
else:
    print("INVALID")
